utf8/sanitize.go

```diff
@@ -0,0 +1,38 @@
+package utf8
+
+import "unicode/utf8"
+
+const replacementChar = string('?')
+
+var replacementCharByte = []byte(replacementChar)[0]
+
+func init() {
+	if len([]byte(replacementChar)) != 1 {
+		panic("replacementChar must be a single-byte character")
+	}
+}
```

Comment:
> [init is called after all the variable declarations in the package](https://go.dev/doc/effective_go#:~:text=init%20is%20called%20after%20all%20the%20variable%20declarations%20in%20the%20package)
This might not work.

Reply:
It works fine. The variables are first initialized in place and then `init` is called.

In fact if you do this you'll get a `panic`:

```go
const replacementChar = string(utf8.RuneError)

var replacementCharByte = []byte(replacementChar)[0]

func init() {
	if len([]byte(replacementChar)) != 1 {
		panic("replacementChar must be a single-byte character")
	}
}
```

I put it in place as a safety mechanism to make sure people don't use a replacement character that is more than one byte in length.

Reply:
Yeah, that's what I mentioned. init is called post variable creation. I thought you added a check to make sure that `var replacementCharByte = []byte(replacementChar)[0]` is valid since we are accessing 0th index and later in init we are checking if we can access the 0th index.

Reply:
`var replacementCharByte = []byte(replacementChar)[0]` will fail only if you leave `replacementChar` empty and you'll get a `panic` for that even though it won't come from `init`. The logic in `init` is just to make sure you don't use a longer one and as per my tests it works. What do you think is the problem? I'm not sure I understand what is it that you're proposing :sweat_smile:

Reply:
> I put it in place as a safety mechanism to make sure people don't use a replacement character that is more than one byte in length.


Ok. I got now. You just want to make sure that replacing character is not more then 1 byte.